### ISSUE TITLE
InvalidCastException when using Windows.Security.Cryptography.Core.CryptographicKey

### ISSUE THREAD

**Issue by azchohfi** [microsoft/CsWinRT#362]
This code throws:
```cs
string key = ".....";
IBuffer keyMaterial = CryptographicBuffer.ConvertStringToBinary(key, BinaryStringEncoding.Utf8);
MacAlgorithmProvider mac = MacAlgorithmProvider.OpenAlgorithm(MacAlgorithmNames.HmacSha1);
CryptographicKey cryptoKey = mac.CreateKey(keyMaterial);
```
Exception:
```
Exception thrown at 0x00007FF8308832AC (KernelBase.dll) in Microsoft.Toolkit.Uwp.SampleApp.exe: WinRT originate error - 0x80004002 : 'Unable to cast object of type 'WinRT.IInspectable' to type 'ABI.Windows.Security.Cryptography.Core.ICryptographicKey'.'.
```

**Comment by j0shuams** [comment]
Seems the issue is that a RCW tries to be created for Windows.Security.Cryptography.Core.MacKey, but this type doesn't exist so a general IInspectable object gets returned by the FromAbi call of the CryptograhyKey class. MacKey is the runtime class name information used to make the object returned by FromABI, but the provided type information (CryptographyKey) should be used to construct the object instead. 

MacKey is not a valid type, as seen [here](https://docs.microsoft.com/en-us/uwp/api/windows.security.cryptography.core?view=winrt-19041), it must be used internally to represent CryptographyKey.

@manodasanW is working on passing type information as generics in FromABI, and after our discussion I am assigning him.

**Comment by Scottj1s** [comment]
fixed by #369
